driver/bpf/fillers.h

```diff
@@ -7340,24 +7340,20 @@ FILLER(sys_splice_e, true) {
 }
 
 FILLER(sys_dup_e, true) {
-	/* Parameter 1: oldfd (type: PT_FD) */
-	int32_t oldfd = (int32_t)bpf_syscall_get_argument(data, 0);
-	return bpf_push_s64_to_ring(data, (int64_t)oldfd);
+	/* Parameter 1: fd (type: PT_FD) */
+	int64_t oldfd = (int64_t)(int32_t)bpf_syscall_get_argument(data, 0);
```

Comment:
Double cast?

Reply:
i get the point of zero-ing upper 32 bits, yet this feels really weird :laughing:

Reply:
Yeah agree. That's just the way we deal with the fact that we wants 32 bits file descriptors encoded as 64 bits fields :pensive: